test/jdk/valhalla/valuetypes/InlineTypeConversionTest.java

```diff
@@ -0,0 +1,159 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/*
+ * @test
+ * @summary test method handles with inline type conversion
+ * @run testng/othervm InlineTypeConversionTest
+ */
+
+import java.lang.invoke.*;
+
+import static java.lang.invoke.MethodType.*;
+
+import org.testng.annotations.Test;
+import static org.testng.Assert.*;
+
+public class InlineTypeConversionTest {
+    static inline class Value {
+        Point val;
+        Point.ref ref;
+        Value(Point p1, Point.ref p2) {
+            this.val = p1;
+            this.ref = p2;
+        }
+    }
+
+    static Value narrow(Value.ref v) {
+        return v;
+    }
+
+    static Value.ref widen(Value v) {
+        if (((Object)v) == null) {
+            throw new Error("should never reach here: should be caught by runtime");
+        }
+        return null;
+    }
+
+    static final Value VALUE = new Value(new Point(10,10), new Point(20, 20));
+
+    @Test
+    public static void inlineWidening() throws Throwable {
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        MethodHandle mh1 = lookup.findStatic(InlineTypeConversionTest.class, "narrow", methodType(Value.class, Value.ref.class));
+        MethodHandle mh2 = mh1.asType(methodType(Value.class, Value.class));
+        Object v = mh1.invoke(VALUE);
+        assertEquals(v, VALUE);
+        try {
+            Object v1 = mh1.invoke((Object)null);
+            fail("Expected NullPointerException but not thrown");
+        } catch (NullPointerException e) {}
+
+        try {
+            Object v2 = mh2.invoke((Object)null);
+            fail("Expected NullPointerException but not thrown");
+        } catch (NullPointerException e) {}
+    }
+
+    @Test
+    public static void inlineNarrowing() throws Throwable {
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        MethodHandle mh = lookup.findStatic(InlineTypeConversionTest.class, "widen", methodType(Value.ref.class, Value.class));
+        Object v = mh.invoke(VALUE);
+        assertTrue(v == null);
+        try {
+            Object v1 = mh.invoke((Object)null);
+            fail("Expected NullPointerException but not thrown");
+        } catch (NullPointerException e) {
+            e.printStackTrace();
```

Comment:
Needn't print stack trace here or below after invoking mh2.

Reply:
It's not needed and it's left-over from my debugging.  I'll clean it up when we touch this file.